src/main/java/uk/gov/hmcts/reform/civil/service/docmosis/RespondToWrittenRepresentationGenerator.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.civil.service.docmosis;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.civil.model.CaseData;
+import uk.gov.hmcts.reform.civil.model.docmosis.DocmosisDocument;
+import uk.gov.hmcts.reform.civil.model.docmosis.judgedecisionpdfdocument.JudgeDecisionPdfDocument;
+import uk.gov.hmcts.reform.civil.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.civil.model.documents.DocumentType;
+import uk.gov.hmcts.reform.civil.model.documents.PDF;
+import uk.gov.hmcts.reform.civil.service.documentmanagement.DocumentManagementService;
+import uk.gov.hmcts.reform.civil.utils.DocUploadUtils;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static uk.gov.hmcts.reform.civil.service.docmosis.DocmosisTemplates.RESPOND_FOR_WRITTEN_REPRESENTATION;
+
+@Service
+@RequiredArgsConstructor
+public class RespondToWrittenRepresentationGenerator implements TemplateDataGenerator<JudgeDecisionPdfDocument>  {
+
+    private final DocumentManagementService documentManagementService;
+    private final DocumentGeneratorService documentGeneratorService;
+    private String role;
```

Comment:
role should not be class property. I think it can be passed as argument in getTemplateData